catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -75,24 +99,47 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GmdTransformer.class);
 
-    private static final String COLON = ":";
+    public static final String GCO = "gco";
 
-    private static final GmdMetacardType GMD_METACARD_TYPE = new GmdMetacardType();
+    private static final String GML_PREFIX = "gml:";
 
-    private static final String GCO_COLON = GmdMetacardType.GCO_PREFIX + ":";
+    private static final String GCO_PREFIX = GCO + ":";
 
     private static final String TRANSFORM_EXCEPTION_MSG =
             "Unable to transform from GMD Metadata to Metacard";
 
+    private static final Pattern WHITE_SPACE_PATTER = Pattern.compile("(\\s)+");
+
+    private static final Pattern NEW_LINE_PATTERN = Pattern.compile("(\\n)+");
+
+    private static final String ZERO_HOUR_SUFFIX = "T00:00:00Z";
+
+    private static MetacardType gmdMetacardType;
+
     private final XStream xstream;
 
     private final DataHolder argumentHolder;
 
     private final XMLInputFactory xmlFactory;
 
-    public GmdTransformer() {
+    private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL =
```

Comment:
It looks like you do not clean up your ThreadLocal.  This could cause a memory leak depending if the bundle is restarted or configuration causes the bean to reconstructed.

Are we sure WKTWriter is that expensive to require ThreadLocal?  WKTReader can be slow with the no arg constructor since it creates a new geometry factory but WKTWriter does not have any heavy weight construction that I am aware of.  We usually just create a new WKTWriter as needed.

Here is an example of a class that hooks up the destroy method in blueprint to avoid ThreadLocal memory leaks.

https://github.com/codice/ddf/blob/eaca0caec1f345cd96e800f2aeecc708054aecd1/platform/security/filter/security-filter-login/src/main/java/org/codice/ddf/security/filter/login/LoginFilter.java#L756-L760